spec/schemas/_common.yaml

```diff
@@ -656,6 +656,16 @@ components:
         boost:
           description: Boost value to apply to kNN scores
           type: number
+        method_parameters:
+          type: object
+          x-version-added: '2.16'
+          additionalProperties:
+            type: number
+        rescore:
+          type: object
+          x-version-added: '2.17'
+          additionalProperties:
+            type: number
```

Comment:
if I understand this correctly, the additionalProperties are more of a map whose values will always be of type number? correct?

if the above understanding is correct, then my worry is going forward we can add more parameters and those parameters might not be of number type. Can use some type like object or anything that can allow us to specify number, strings as additional parameter values in future.

@dblock thoughts?

Reply:
If we add them in the future, we can update this with:
```
anyOf:
  type:
    number
    string
    etc.
```

Reply:
@navneet1v you're correct, and @ryanbogan is also correct, if the schema _today_ is a number, leave it as such, and tomorrow it can be extended when it can actually be something else.

Reply:
Yes.